packages/node_modules/@ciscospark/common/test/unit/spec/one-flight.js

```diff
@@ -5,6 +5,43 @@ import {assert} from '@ciscospark/test-helper-chai';
 
 describe(`common`, () => {
   describe(`@oneFlight`, () => {
+    it(`returns new promise on function with undefined keyFactory`, () => {
+      const C = AmpState.extend({
+        @oneFlight({keyFactory: (param1) => param1})
+        funcC1(param1) {
+          return new Promise((resolve) => {
+            process.nextTick(() => {
+              this.spy();
+              resolve(param1);
+            });
+          });
+        },
+
+        @oneFlight({keyFactory: (param2) => param2})
+        funcC2(param2) {
+          return new Promise((resolve) => {
+            this.spy();
+            resolve(param2);
+          });
+        },
+
+        spy() {
+          return true;
+        }
+      });
+
+      const c = new C();
+      sinon.spy(c, `spy`);
+
+      return Promise.all([
+        c.funcC1(),
+        c.funcC2()
+      ])
+        .then(() => {
+          assert.calledTwice(c.spy);
+        });
+    });
+
```

Comment:
Please add another test for the original scenario of calling the same method twice with an undefined key factory result.